mixer/pkg/lang/compiler/compiler.go

```diff
@@ -366,6 +374,156 @@ func (g *generator) generateEq(f *ast.Function, depth int) {
 	}
 }
 
+func (g *generator) generateLt(f *ast.Function, depth int) {
+	exprType := g.evalType(f.Args[0])
+	g.generate(f.Args[0], depth+1, nmNone, "")
+
+	var constArg1 interface{}
+	if f.Args[1].Const != nil {
+		constArg1 = f.Args[1].Const.Value
+	} else {
+		g.generate(f.Args[1], depth+1, nmNone, "")
+	}
+
+	switch exprType {
+
+	case il.String:
+		if constArg1 != nil {
+			g.builder.ALTString(constArg1.(string))
+		} else {
+			g.builder.LTString()
+		}
+
+	case il.Integer:
+		if constArg1 != nil {
+			g.builder.ALTInteger(constArg1.(int64))
+		} else {
+			g.builder.LTInteger()
+		}
+
+	case il.Double:
+		if constArg1 != nil {
+			g.builder.ALTDouble(constArg1.(float64))
+		} else {
+			g.builder.LTDouble()
+		}
+
+	default:
+		g.internalError("less than for type not yet implemented: %v", exprType)
+	}
+}
+
+func (g *generator) generateGt(f *ast.Function, depth int) {
+	exprType := g.evalType(f.Args[0])
+	g.generate(f.Args[0], depth+1, nmNone, "")
+
+	var constArg1 interface{}
+	if f.Args[1].Const != nil {
+		constArg1 = f.Args[1].Const.Value
+	} else {
+		g.generate(f.Args[1], depth+1, nmNone, "")
+	}
+
+	switch exprType {
+
+	case il.String:
+		if constArg1 != nil {
+			g.builder.AGTString(constArg1.(string))
+		} else {
+			g.builder.GTString()
+		}
+	case il.Integer:
+		if constArg1 != nil {
+			g.builder.AGTInteger(constArg1.(int64))
+		} else {
+			g.builder.GTInteger()
+		}
+	case il.Double:
+		if constArg1 != nil {
+			g.builder.AGTDouble(constArg1.(float64))
+		} else {
+			g.builder.GTDouble()
+		}
+
+	default:
+		g.internalError("greater than for type not yet implemented: %v", exprType)
+	}
+}
+
+func (g *generator) generateGe(f *ast.Function, depth int) {
+	exprType := g.evalType(f.Args[0])
+	g.generate(f.Args[0], depth+1, nmNone, "")
+
+	var constArg1 interface{}
+	if f.Args[1].Const != nil {
+		constArg1 = f.Args[1].Const.Value
+	} else {
+		g.generate(f.Args[1], depth+1, nmNone, "")
+	}
+
+	switch exprType {
+
+	case il.String:
+		if constArg1 != nil {
+			g.builder.AGEString(constArg1.(string))
+		} else {
+			g.builder.GEString()
+		}
+	case il.Integer:
+		if constArg1 != nil {
+			g.builder.AGEInteger(constArg1.(int64))
+		} else {
+			g.builder.GEInteger()
+		}
+	case il.Double:
+		if constArg1 != nil {
+			g.builder.AGEDouble(constArg1.(float64))
+		} else {
+			g.builder.GEDouble()
+		}
+	default:
+		g.internalError("greater than for type not yet implemented: %v", exprType)
+	}
+}
+
+func (g *generator) generateLe(f *ast.Function, depth int) {
+	exprType := g.evalType(f.Args[0])
+	g.generate(f.Args[0], depth+1, nmNone, "")
+
+	var constArg1 interface{}
+	if f.Args[1].Const != nil {
+		constArg1 = f.Args[1].Const.Value
+	} else {
+		g.generate(f.Args[1], depth+1, nmNone, "")
+	}
+
+	switch exprType {
+
+	case il.String:
+		if constArg1 != nil {
+			g.builder.ALEString(constArg1.(string))
```

Comment:
This may panic if the argument types are mismatched, e.g. (1 < "1"). I think there needs to be some type checking applied to handle that, or some error-tolerant approach defined (via type conversion).

Reply:
I added a test for this and it seems to be safe and return a compile error.
`==` works in the same way.
I'll add a test for this case